src/xma/src/xmaapi/xmafilter.cpp

```diff
@@ -243,40 +196,58 @@ xma_filter_session_create(XmaFilterProperties *filter_props)
     priv1->kernel_execbos.reserve(num_execbo);
     priv1->num_execbo_allocated = num_execbo;
     if (xma_core::create_session_execbo(priv1, num_execbo, XMA_FILTER_MOD) != XMA_SUCCESS) {
-        //Release singleton lock
-        g_xma_singleton->locked = false;
         free(filter_session->base.plugin_data);
         free(filter_session);
         delete priv1;
         return nullptr;
     }
 
-    rc = filter_session->filter_plugin->init(filter_session);
-    if (rc) {
-        xma_logmsg(XMA_ERROR_LOG, XMA_FILTER_MOD,
-                   "Initalization of filter plugin failed. Return code %d\n",
-                   rc);
-        //Release singleton lock
-        g_xma_singleton->locked = false;
-        free(filter_session->base.plugin_data);
-        free(filter_session);
-        delete priv1;
-        return nullptr;
+    //Obtain lock only for a) singleton changes & b) kernel_info changes
```

Comment:
Add comment the start and stop of a mutex....or....place the mutex in its own scoping brackets.